src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -320,7 +320,9 @@ export class ExtensionsListView extends CollapsibleView {
 				}
 
 				return this.extensionsWorkbenchService.queryGallery(assign(options, { names, pageSize: names.length }))
-					.then(pager => new PagedModel(pager));
+					.then(pager => this.addRecommendationInfo(new PagedModel(pager || []), {
+						source: 'fileType'
+					}));
```

Comment:
Please see if you can carry the `source of query` information into telemetry data of IGalleryExtension.